src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJModelFactory.java

```diff
@@ -22,11 +28,43 @@ public IntelliJModelFactory(@NotNull Project project) {
   public Course createCourse(@NotNull String name,
                              @NotNull List<Module> modules,
                              @NotNull Map<String, String> requiredPlugins) {
-    return new IntelliJCourse(name, modules, requiredPlugins, project);
+    IntelliJCourse course = new IntelliJCourse(name, modules, requiredPlugins, project);
+    // Add a module change listener with the created course instance to the project
+    project.getMessageBus().connect().subscribe(ProjectTopics.MODULES, new ModuleListener() {
```

Comment:
can this chained call can be extracted into a separate method (then there could be a call to it on line 43 as well)?